lib/src/revset.rs

```diff
@@ -880,34 +891,46 @@ fn parse_symbol_rule(
     state: ParseState,
 ) -> Result<Rc<RevsetExpression>, RevsetParseError> {
     let first = pairs.next().unwrap();
-    match first.as_rule() {
+    let symbol = match first.as_rule() {
         Rule::identifier => {
             let name = first.as_str();
             if let Some(expr) = state.locals.get(name) {
-                Ok(expr.clone())
+                return Ok(expr.clone());
             } else if let Some((id, defn)) = state.aliases_map.get_symbol(name) {
                 let locals = HashMap::new(); // Don't spill out the current scope
-                state.with_alias_expanding(id, &locals, first.as_span(), |state| {
+                return state.with_alias_expanding(id, &locals, first.as_span(), |state| {
                     parse_program(defn, state)
-                })
+                });
             } else {
-                Ok(RevsetExpression::symbol(name.to_owned()))
+                name.to_owned()
             }
         }
-        Rule::literal_string => {
-            return Ok(RevsetExpression::symbol(
-                first
-                    .as_str()
-                    .strip_prefix('"')
-                    .unwrap()
-                    .strip_suffix('"')
-                    .unwrap()
-                    .to_owned(),
-            ));
-        }
+        Rule::literal_string => first
+            .as_str()
+            .strip_prefix('"')
+            .unwrap()
+            .strip_suffix('"')
+            .unwrap()
+            .to_owned(),
         _ => {
             panic!("unexpected symbol parse rule: {:?}", first.as_str());
         }
+    };
+    if symbol.ends_with('@') {
+        let workspace_id = if symbol == "@" {
+            if let Some(ctx) = state.workspace_ctx {
+                ctx.workspace_id.clone()
+            } else {
+                return Err(RevsetParseError::new(
+                    RevsetParseErrorKind::WorkingCopyWithoutWorkspace,
+                ));
+            }
+        } else {
+            WorkspaceId::new(symbol.strip_suffix('@').unwrap().to_string())
+        };
+        Ok(RevsetExpression::working_copy(workspace_id))
+    } else {
```

Comment:
Unfortunately, interpreting symbol here breaks queries like `author("martinvonz@")`.

I think alias expansion and string argument parsing should be performed at higher (closer-to-AST) level, but they aren't right now. Perhaps, the easiest workaround is to set "don't interpret symbol" flag to `ParseState` by `parse_function_argument_to_string_pattern`/`as_literal()`.

Reply:
Good catch! I'm not sure how I think a plain `"@"` should be interpreted. We currently interpret it as `@`, but maybe we should interpret it as a (non-special) symbol instead.

I'm starting to think that this PR won't help you at all because you're going to do it in a better way anyway. I really don't mind dropping it, but I'm also happy to add that workaround if it you think it would help you.

Reply:
> I'm not sure how I think a plain `"@"` should be interpreted. We currently interpret it as `@`, but maybe we should interpret it as a (non-special) symbol instead.

I agree. Quoted `"@"` can be interpreted as a branch or tag name.

I haven't started any work for #2101 or #2095 yet. Maybe we can go one step towards them by restricting `@` and `workspace@` to be unquoted (i.e. `Rule::identifier`)? `author("@")` requires quoted string instead (if the pattern ends with `@`.)

I think this PR is nice, and I'll probably add `branch@remote` handling on top.

Reply:
> Maybe we can go one step towards them by restricting `@` and `workspace@` to be unquoted (i.e. `Rule::identifier`)? `author("@")` requires quoted string instead (if the pattern ends with `@`.)

Sounds good. I'll try to update this PR to do that.

Reply:
I think that's now done. Let me know if it's not what you had in mind.